dev/buildtool/spin_commands.py

```diff
@@ -330,21 +335,28 @@ def promote_spin(self, repository):
     # and minor versions so fixes (thus patch version) are decoupled between
     # the two.
     patch = '0' # Patch is reset on a new Gate major or minor.
-    tag_matcher = re.compile(r'version-{maj}.{min}.(\d+)'
-                             .format(maj=gate_major, min=gate_min))
+    tag_pattern = r'version-{maj}.{min}.(\d+)'.format(maj=gate_major, min=gate_min)
+    tag_matcher = re.compile(tag_pattern)
```

Comment:
I think you want a helper function here and above that returns the highest tag (or semver) for a given <tag_pattern>. If it doesnt have one then you create a new semver at patch 0. Otherwise you  use semver.next(semver.PATCH_INDEX)

Reply:
:+1: refactored.